streaming/stream_transfer_task.cc

```diff
@@ -237,18 +237,23 @@ future<> stream_transfer_task::execute() {
     }).then([this, id, plan_id] {
         _mutation_done_sent = true;
         sslog.debug("[Stream #{}] GOT STREAM_MUTATION_DONE Reply from {}", plan_id, id.addr);
-    }).handle_exception([this, plan_id, cf_id, id] (std::exception_ptr ep) {
+    }).handle_exception_type([this, plan_id, cf_id, id] (const replica::no_such_column_family& e) {
         // If the table is dropped during streaming, we can ignore the
         // errors and make the stream successful. This allows user to
         // drop tables during node operations like decommission or
         // bootstrap.
-        if (!session->manager().db().column_family_exists(cf_id)) {
```

Comment:
If we really want to check if any of the shard knows the table is dropped. We can query on all shards and return true if any of them knows the table is dropped.